### PR TITLE
Add otel toleration for dedicated nodes

### PR DESCRIPTION
**Pull request by lukaszgarstecki**

### CODE REVIEW

File: apps_otelcol.tf [hunk 1]
```diff
@@ -186,6 +186,12 @@ resource "kubernetes_daemonset" "otel_agent" {
             }
           }
         }
+
+        toleration {
```
Comment (jakubfijalkowski) [inline apps_otelcol.tf]: Too specific. This is much too much LNCD-dependent, we need to pass `toleration`s blocks  as a variable to the module.

Comment (lukaszgarstecki) [reply apps_otelcol.tf]: I thought this is LNCD-specific repository, if you want I'll adjust it

Comment (jakubfijalkowski) [reply apps_otelcol.tf]: It's not _that_ specific - see external projects. :)

File: variables.tf [hunk 2]
```diff
@@ -138,7 +138,13 @@ variable "opentelemetry" {
         cpu    = string,
         memory = string,
       }),
-    })
+    }),
+    tolerations = list(object({
```
Comment (jakubfijalkowski) [inline variables.tf:142]: https://www.terraform.io/language/expressions/type-constraints#experimental-optional-object-type-attributes pls

Comment (jakubfijalkowski) [inline variables.tf]: https://www.terraform.io/language/expressions/type-constraints#experimental-optional-object-type-attributes pls (and `value` in there also)

**Comment by lukaszgarstecki** [comment]
@lukaszgarstecki can't into terraform :upside_down_face:

**Review by Saancreed — APPROVED** [review]

**Review by jakubfijalkowski — APPROVED** [review]

**Pull request merged**